membership.lua

```diff
@@ -28,39 +26,39 @@ local _sock = nil
 local _sync_trigger = fiber.cond()
 local _ack_trigger = fiber.cond()
 local _ack_cache = {}
-
 local _resolve_cache = {}
+
 local function resolve(uri)
     checks('string')
 
     local member = members.get(uri)
-    if member and member.status == opts.ALIVE then
-        local _cached = _resolve_cache[uri]
-        if _cached then
-            return unpack(_cached)
-        end
+    local cached = _resolve_cache[uri]
+    if cached and member and member.status == opts.ALIVE then
+        return cached
     end
 
     local parts = uri_tools.parse(uri)
     if not parts then
-        return nil, nil, 'parse error'
+        return nil, 'parse error'
     end
 
-    local hosts = socket.getaddrinfo(parts.host, parts.service, {family='AF_INET', type='SOCK_DGRAM'})
-    if hosts == nil or #hosts == 0 then
-        return nil, nil, 'getaddrinfo failed'
+    local addrinfo = socket.getaddrinfo(
```

Comment:
You should check the second value of getaddrinfo.

See https://github.com/tarantool/doc/issues/1431 (Return error message as the second return value.)

If it's empty we should return `'getaddrinfo failed'`